### DISCUSSION TITLE
adding Custom zeek Script doesn't show up in loaded_scripts

### DISCUSSION THREAD

**Discussion by rana97ahmad** [Security-Onion-Solutions/securityonion#4499] [Unsupported Versions]
I am trying to add a custom zeek script and followed the steps in https://docs.securityonion.net/en/latest/zeek.html#custom-scripts
However, the script is not loaded in the `loaded_scripts.log`
Below are the files in the` /opt/so/saltstack/local/salt/zeek/policy/custom/ASTERIX34` directory

![image](https://user-images.githubusercontent.com/44316352/121771572-ca91d100-cb78-11eb-92b9-557f5be2d388.png)

the `__load__.zeek` has the following line:
`@load ./ASTERIX34`

and this is the beginning of my script:
```
## ASTERIX protocol analyzer for category 34

## ----------------------- GLOBAL VARIABLES --------------------------
# Define a namespace
module ASTERIX34;
# Turn on UDP content delivery
redef udp_content_deliver_all_orig = T;
redef udp_content_deliver_all_resp = T;

```

and I added `-custom/ASTERIX34`  under the load section in /opt/so/saltstack/local/pillar/zeek/init.sls file

and then I restarted zeek using ` sudo so-zeek-restart`

What could i be missing or doing wrong?

**Comment by weslambert** [comment]
> I am trying to add a custom zeek script and followed the steps in https://docs.securityonion.net/en/latest/zeek.html#custom-scripts
> However, the script is not loaded in the `loaded_scripts.log`
> Below are the files in the` /opt/so/saltstack/local/salt/zeek/policy/custom/ASTERIX34` directory
> 
> ![image](https://user-images.githubusercontent.com/44316352/121771572-ca91d100-cb78-11eb-92b9-557f5be2d388.png)
> 
> the `__load__.zeek` has the following line:
> `@load ./ASTERIX34`
> 
> and this is the beginning of my script:
> 
> ```
> ## ASTERIX protocol analyzer for category 34
> 
> ## ----------------------- GLOBAL VARIABLES --------------------------
> # Define a namespace
> module ASTERIX34;
> # Turn on UDP content delivery
> redef udp_content_deliver_all_orig = T;
> redef udp_content_deliver_all_resp = T;
> ```
> 
> and I added `-custom/ASTERIX34` under the load section in /opt/so/saltstack/local/pillar/zeek/init.sls file
> 
> and then I restarted zeek using ` sudo so-zeek-restart`
> 
> What could i be missing or doing wrong?

Is this a standalone or distributed deployment?

Should `-custom/ASTERIX34` be `- custom/ASTERIX34`?

What are the permissions for the custom files?

Are there any clues in reporter.log?

**Comment by peasead** [comment]
Here's what I just did.

1. From `/opt/so/saltstack/local/salt/zeek/policy/custom/`, I ran `git clone https://github.com/jsiwek/zeek-cryptomining`
2. Copied `/opt/so/saltstack/default/pillar/zeek/init.sls` to `/opt/so/saltstack/local/pillar/zeek/init.sls`
3. Added the following to `/opt/so/saltstack/local/pillar/zeek/init.sls`

```
local:
  '@load':
      ...
      - custom/zeek-cryptomining
```
4. Validated all files/directories were owned by `socore`
5. Ran `sudo so-zeek-restart`
6. Ran `grep zeek-cryptomining /nsm/zeek/logs/current/loaded_scripts.log`
```
{"name":"  /opt/zeek/share/zeek/policy/custom/zeek-cryptomining/__load__.zeek"}
{"name":"    /opt/zeek/share/zeek/policy/custom/zeek-cryptomining/mining.zeek"}
```
I haven't tested to see if it's working, but seemingly, it is loaded.
